Aquality.WinAppDriver/src/Aquality.WinAppDriver/Applications/WindowHandleApplicationFactory.cs

```diff
@@ -0,0 +1,53 @@
+﻿using Aquality.WinAppDriver.Configurations;
+using Microsoft.Extensions.DependencyInjection;
+using OpenQA.Selenium.Appium;
+using OpenQA.Selenium.Appium.Windows;
+using OpenQA.Selenium.Remote;
+using System;
+
+namespace Aquality.WinAppDriver.Applications
+{
+    public class WindowHandleApplicationFactory : ApplicationFactory
+    {
+        private const string WindowHandleCapability = "appTopLevelWindow";
+        private const string AppNameCapability = "app";
+        private readonly Uri driverServerUri;
+        private readonly Func<WindowsDriver<WindowsElement>, string> getWindowHandleFunction;
+        private readonly bool isRemote;
+
+        public WindowHandleApplicationFactory(Uri driverServerUri, IServiceProvider serviceProvider, Func<WindowsDriver<WindowsElement>, string> getWindowHandleFunction)
+            : base(serviceProvider)
+        {
+            this.driverServerUri = driverServerUri;
+            this.getWindowHandleFunction = getWindowHandleFunction;
+            isRemote = serviceProvider.GetRequiredService<IApplicationProfile>().IsRemote;
+        }
+
+        public override Application Application
+        {
+            get
+            {
+                LocalizedLogger.Info(isRemote ? "loc.application.driver.service.remote" : "loc.application.driver.service.local.start", driverServerUri);
```

Comment:
please create variable for `isRemote ? "loc.application.driver.service.remote" : "loc.application.driver.service.local.start"`

Reply:
Maybe it makes sense to leave only one message? Probably `driverServerUri` has different values for remote and local session and we can identify it by its value.

Reply:
actually it is configurable and could be the same

Reply:
comment#1 fixed